types/index.d.ts

```diff
@@ -1,648 +1,62 @@
-         * @returns the auto-incrementing unique id of this IndoorMapEntityInformation object.
-         */
-        getNativeId(): number;
-    }
-    
-    function indoorMapEntityInformation(indoorMapId: IndoorMapId): IndoorMapEntityInformation;
-
-}
-
-/* Wrld.indoorMapFloorOutlines */
-
-declare namespace indoorMapFloorOutlines {
-
-    interface IndoorMapFloorOutlineInformationLoadedEvent extends Event { indoorMapFloorOutlineInformation: indoorMapFloorOutlines.IndoorMapFloorOutlineInformation }
-    type IndoorMapFloorOutlineInformationLoadedEventHandler = EventHandler<IndoorMapFloorOutlineInformationLoadedEvent>;
-
-    type EventType = "indoormapflooroutlineinformationloaded";
-
-    class IndoorMapFloorOutlines {
-        on(type: "indoormapflooroutlineinformationloaded", fn: IndoorMapFloorOutlineInformationLoadedEventHandler): void;
-        once(type: "indoormapflooroutlineinformationloaded", fn: IndoorMapFloorOutlineInformationLoadedEventHandler): void;
-        off(event: EventType, handler: (e: Event) => void): this;
-    }
-
-    class IndoorMapFloorOutlineInformation {
-        constructor(indoorMapId: IndoorMapId, indoorMapFloorId: IndoorMapFloorId);
-        getIndoorMapId(): IndoorMapId;
-        getIndoorMapFloorId(): IndoorMapFloorId;
-        getIndoorMapFloorOutlinePolygons(): IndoorMapFloorOutlinePolygon[];
-        getIsLoaded(): boolean;
-        getId(): number;
-        addTo(map: Map): this;
-        remove(): this;
-    }
-    
-    function indoorMapFloorOutlineInformation(indoorMapId: IndoorMapId, indoorMapFloorId: IndoorMapFloorId): IndoorMapFloorOutlineInformation;
-
-    class IndoorMapFloorOutlinePolygon {
-        getOuterRing(): IndoorMapFloorOutlinePolygonRing;
-        getInnerRings(): IndoorMapFloorOutlinePolygonRing[];
-    }
-
-    class IndoorMapFloorOutlinePolygonRing {
-        getLatLngPoints(): L.LatLng[];
-    }
-}
-
-/* Wrld */
-
-declare function map(element: HTMLElement | string, apiKey: string, options?: MapOptions): Map;
-declare function marker(latLng: L.LatLngExpression, options?: MarkerOptions): Marker;
-declare function popup(options?: PopupOptions, source?: L.Layer): Popup;
-declare function polygon(latlngs: L.LatLngTuple[] | L.LatLngTuple[][], options?: PolygonOptions): Polygon;
-declare function polyline(latlngs: L.LatLngExpression[], options?: PolylineOptions): Polyline;
-declare function prop(name: string, geometryId: string, location: L.LatLngExpression, options?: props.PropOptions): props.Prop;
-declare function heatmap(pointData: Heatmap.PointData[], options?: HeatmapOptions): Heatmap;
+// Types
+import type L from "leaflet";
+import type { WrldEvent, EventHandler } from "./event";
+import type { Color } from "./color";
+import type { ElevationMode } from "./elevationMode";
+import type { Vector3, Vector4 } from "./vector";
+// Imports
+import Map from "./map";
+import Marker from "./marker";
+import props from "./props";
```

Comment:
The case-inconsistency here is bothering me but it's not even that all namespaces are lower -cased, so I don't think there is much we can do?

Reply:
Took me a while to have a look at that, I thought that it was because of wrld.js, but actually it turns out that's just how Calum defined the namespaces to begin with. At this point it will have many breaking changes, might as well just wait until we rewrite `index.js` TypeScript, that will ensure the naming matches the actual implementation, and it's gonna be worth the re-write of all the imports in the consumers, not to mention that it will break the public APIs.

If we go on and do that, we should probably up wrld.js by one major version.

Reply:
Yep, leave it as is for now!